File: programmers/42586.js
function solution(progresses, speeds) {
  var answer = [0];
  // 인덱스의 day를 구한다.
  let days = [];

  for (let i in progresses) {
    // 반올림 말고 올림처리
    let day = Math.ceil((100 - progresses[i]) / speeds[i]);
    days.push(day);
  }
  let max = days[0]; // 최대 일자를 첫번째 값으로 초기화 , 이후 비교하며 교체

  console.log(days);

  // days 배열 순회
  for (let i = 0, j = 0; i < days.length; i++) {
    // max보다 작으면 count 증가
    if (days[i] <= max) {
      answer[j] += 1;
    } else {
      // 최댓값 교체 후
      max = days[i];
      // count 증가해서 answer 배열에 추가
      answer[++j] = 1;
    }
  }

  return answer;
}

File: programmers/42586.py
from collections import deque

def solution(progresses, speeds):
    # day(일수), deployCount(배포 개수) 선언
    day = 0
    deployCount = 0
    answer = []

    # Queue 생성
    progressQueue = deque()

    # Queue 초기화 (progress, speed)
    for progress, speed in zip(progresses, speeds):
        progressQueue.append((progress, speed))

    while progressQueue:
        # day 증가
        day += 1

        # Queue가 비어있지 않고 진행도가 100이 넘는 경우
        while progressQueue and progressQueue[0][0] + (day * progressQueue[0][1]) >= 100:
            deployCount += 1
            progressQueue.popleft()

        # 해당 일에 배포된 개수가 0이 아닌 경우
        if deployCount != 0:
            answer.append(deployCount)
            deployCount = 0

    return answer

        
solution([93, 30, 55],[1, 30, 5])